lib/StripeResource.js

```diff
@@ -241,28 +255,37 @@ StripeResource.prototype = {
     var self = this;
     var requestData;
 
-    if (self.requestDataProcessor) {
-      requestData = self.requestDataProcessor(method, data, options.headers);
-    } else {
-      requestData = utils.stringifyRequestData(data || {});
-    }
+    function makeRequestWithData(error, data) {
+      var apiVersion;
+      var headers;
 
-    var apiVersion = this._stripe.getApiField('version');
+      if (error) {
+        var handleError = self._errorHandlerStream(callback);
```

Comment:
I'd rather see this FileUpload-specific error generation/handling pushed to `FileUploads.js`, and have this just be more generic here so it can handle future cases, i.e. just:

    if (error) {
      return callback(error);
    }

At this point in the code, we don't (or shouldn't) really care what kind of error this is, we should just expose it in the `callback`.

Reply:
updated